resources/Release Checklist.md

```diff
@@ -1,12 +0,0 @@
-
-# Release Checklist
-
-* [ ] Check that CI passed
-
-* [ ] Update version number
-
-* [ ] Update `fl_model/configuration/schema.json` for major versions
-
-* [ ] Update `LATEST_API_VERSION` from `fl_model/consts.py` for major versions
-
-* [ ] Write release notes and release
```

Comment:
I'd prefer to keep this here for the time being, since I want to ensure that I don't miss any important steps when releasing new versions.